src/ensembl/src/shared/components/instant-download/instant-download-transcript/InstantDownloadTranscriptVisualisation.tsx

```diff
@@ -128,17 +136,36 @@ const InstantDownloadTranscriptVisualisation = (props: Props) => {
     );
   });
 
-  const proteinSegments = times(exonsCount, (index) => {
-    const isOuterSegment = index === 0 || index === exonsCount - 1;
-    return (
-      <rect
-        key={index}
-        className={styles.protein}
-        x={index === 0 ? halfExonWidth : index * (exonWidth + halfExonWidth)}
-        y={exonHeight + verticalGap}
-        width={isOuterSegment ? halfExonWidth : exonWidth}
-        height={proteinHeight}
-      />
+  const proteinSegments = times(exonsCount, (si) => {
+    const isOuterSegment = si === 0 || si === exonsCount - 1;
+
+    const calculateXPosition = (segmentIndex: number, blockIndex: number) => {
```

Comment:
doesn't this function want to be outside of `proteinSegments`? You are effectively re-declaring it seven times.

Reply:
That's done. 
I did declare it outside initially but I was a bit skeptical about naming it.